intellij/src/saros/intellij/editor/ProjectAPI.java

```diff
@@ -56,7 +57,9 @@ public static boolean isOpen(@NotNull Project project, @NotNull Document documen
    * @return whether there is an open text editor for the given file
    */
   public static boolean isOpenInTextEditor(@NotNull Project project, @NotNull VirtualFile file) {
-    FileEditor[] fileEditors = getFileEditorManager(project).getEditors(file);
+    FileEditorManager fileEditorManager = getFileEditorManager(project);
+
+    FileEditor[] fileEditors = EDTExecutor.invokeAndWait(() -> fileEditorManager.getEditors(file));
```

Comment:
In contrast to the other methods you wrapped in an `invokeAndWait` the method `getEditors` has [no comment](https://github.com/JetBrains/intellij-community/blob/5ce01ec91f0eae2b44a0f6e712eb219863967ce9/platform/platform-api/src/com/intellij/openapi/fileEditor/FileEditorManager.java#L122) "Must be called from <a href="https://docs.oracle.com/javase/tutorial/uiswing/concurrency/dispatch.html">EDT</a>." Is it necessary to wrap this method?

Reply:
Yes, it is not stated in the javadoc. But in the actual [implementation](https://github.com/JetBrains/intellij-community/blob/5ce01ec91f0eae2b44a0f6e712eb219863967ce9/platform/platform-impl/src/com/intellij/openapi/fileEditor/impl/FileEditorManagerImpl.java#L1312) of the method, it still very much checks for it being run on the EDT.